src/SaslMechanism.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4 foldmethod=marker: */
+// +---------------------------------------------------------------------------
+// | SWAN [ $_SWANBR_SLOGAN_$ ]
+// +---------------------------------------------------------------------------
+// | Copyright $_SWANBR_COPYRIGHT_$
+// +---------------------------------------------------------------------------
+// | Version  $_SWANBR_VERSION_$
+// +---------------------------------------------------------------------------
+// | Licensed ( $_SWANBR_LICENSED_URL_$ )
+// +---------------------------------------------------------------------------
+// | $_SWANBR_WEB_DOMAIN_$
+// +---------------------------------------------------------------------------
+
+namespace Kafka;
+
+/**
++------------------------------------------------------------------------------
+* Kafka sasl mechanism provider interface
++------------------------------------------------------------------------------
+*
+* @package
+* @version $_SWANBR_VERSION_$
+* @copyright Copyleft
+* @author $_SWANBR_AUTHOR_$
++------------------------------------------------------------------------------
+*/
+
+interface SaslMechanism
+{
+    // {{{ functions
+    // {{{ public function __construct()
+    
+    /**
+     *
+     * __construct
+     *
+     * @access public
+     * @return void
+     */
+    public function __construct(array $options);
+
+    // }}}
+    // {{{ public function authenticate()
+    
+    /**
+     *
+     * sasl authenticate
+     *
+     * @access public
+     * @return void
+     */
+    public function authenticate(CommonSocket $socket);
+
+    // }}}
+    // {{{ public function getMechanismName()
+    
+    /**
+     *
+     * get sasl authenticate mechanism name
+     *
+     * @access public
+     * @return string
+     */
+    public function getMechanismName();
```

Comment:
`: string` missing

Reply:
Also, is this something relevant for the public interface of the object? I'd see this as an abstract protected method

Reply:
At present, this function used in `handShake` , define it in abstract class not problem, but I think possible useful in future.